### PR TITLE
Feature/add lom commands

### LINKED ISSUE #770: lights out management
It would by quite cool to be able to use the lights out management functions (PowerON, PowerOFF, Reset) that are available on the MacPro and the M1 Mac Mini with 10 Gigabit Ethernet port.

### What version of `micromdm` are you using?
v1.8

### What `micromdm` command did you run?
I did not find the necessary support for LOMDeviceRequestCommand and LOMSetupRequestCommand in the source code, so I did not even try to run a command.

### What did you expect to see?
An implementation of these commands along with caller script for the api.

### What did you see instead?
Commands not yet implemented.

A point to start is https://9to5mac.com/2020/06/24/whats-new-with-apple-device-management/ and  https://developer.apple.com/documentation/devicemanagement/lightsoutmanagementlom.

### PR DESCRIPTION
**Pull request by hrgbcxd**
add possibility to send lom commands

https://github.com/micromdm/micromdm/issues/770

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -21,7 +21,7 @@ require (
 	go.mozilla.org/pkcs7 v0.0.0-20210826202110-33d05740a352
 	golang.org/x/crypto v0.0.0-20211108221036-ceb1ce70b4fa
 	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
-	google.golang.org/protobuf v1.27.1
+	google.golang.org/protobuf v1.28.0
```
Comment (jessepeterson) [inline go.mod:24]: Could you run `go mod tidy`?

File: mdm/internal/checkinproto/checkin.pb.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.26.0
-// 	protoc        v3.18.1
+// 	protoc-gen-go v1.28.0
+// 	protoc        v3.15.8
```
Comment (jessepeterson) [inline mdm/internal/checkinproto/checkin.pb.go:4]: Go protobuf was updated but it looks like you're running an older version of protoc. Would you mind upgrading that?

**Review by jessepeterson — CHANGES_REQUESTED** [review]
Overall this looks okay in my quick review! What I'd love to see is an end-to-end test in `mdm_command_test.go` though. Could you provide that (as well as the other mentioned things).

**Comment by jessepeterson** [comment]
Also - adding some tools scripts would be useful, too. :)

File: mdm/internal/checkinproto/checkin.pb.go (hunk 2 above)
Comment (hrgbcxd) [reply mdm/internal/checkinproto/checkin.pb.go]: [upgrade protoc to 3.21.2](https://github.com/micromdm/micromdm/pull/839/commits/2dc7b8d5c51b7a7f6d6f06b778c2ba069f045c33)

upgraded to latest protoc

File: go.mod (hunk 1 above)
Comment (hrgbcxd) [reply go.mod:24]: [run go mod tidy](https://github.com/micromdm/micromdm/pull/839/commits/551b401dad89259f14760a83f0c7d772a84b7026)

ran go mod tidy

**Comment by hrgbcxd** [comment]
[add lom_reset to tools scripts](https://github.com/micromdm/micromdm/pull/839/commits/c762bd9ce7df47068d685e5d75a5503ce81e7182)

added shell script as example call.

[add end2end test for LOMDeviceRequest](https://github.com/micromdm/micromdm/pull/839/commits/452fa1678e2aa8d349447b7a230d85c688a90f78)

added e2e test case for LOMDeviceRequest

I can run an actual test tomorrow if all this still works, because in [add missing underscore changes](https://github.com/micromdm/micromdm/pull/839/commits/6156c3eaae336d59f840d9772d5cb741f84eff4f) I changed a generated file which now got changed back to camel case.

**Review by jessepeterson — APPROVED** [review]
Looks like a bunch of non-related protobufs were ran, too. THat's fun, but unnecessary. :)

This lgtm. Just fix the linter and we can merge. :)

**Comment by twentyNoise** [comment]
@hrgbcxd

Hi, I'm trying to get LOM working with MicroMDM and saw your comment that "the lom reset works." Could you share what hardware you tested on? Specifically did it work on a Mac Pro 2019 (T2) or Mac mini? Any special certificate or network configuration? I have extensive logs and am trying to determine if this is a firmware bug. Thanks!

**Pull request merged**